termprogressbar/progressbar_test.go

```diff
@@ -0,0 +1,65 @@
+package termuiprogressbar_test
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/hpcloud/termui/termprogressbar"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestVisible tests if the progressbar is visible if it is initialized with visible=true
+func TestVisible(t *testing.T) {
+	assert := assert.New(t)
+
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	bar := termuiprogressbar.NewProgressBar(10, true)
+	bar.Start()
+	bar.FinishPrint("")
+
+	outC := make(chan string)
+
+	go func() {
+		var buf bytes.Buffer
+		io.Copy(&buf, r)
+		outC <- buf.String()
+	}()
+
+	w.Close()
+	defer func() { os.Stdout = old }()
```

Comment:
The defer should live next to the set (so if we panic right after for some reason it gets picked up).  The point is to let the reader not have to think about checking for it later, not to replace the `Close()` call at the same spot :)